connexion/apis/abstract.py

```diff
@@ -82,17 +51,26 @@ def __init__(self, specification, jsonifier, base_url=None, arguments=None,
         :param pythonic_params: When True CamelCase parameters are converted to snake_case and an underscore is appended
         to any shadowed built-ins
         :type pythonic_params: bool
+        :param options: New style options dictionary.
+        :type options: dict | None
+        :param old_style_options: Old style options support for backward compatibility. Preference is
+                                  what is defined in `options` parameter.
         """
         self.debug = debug
         self.validator_map = validator_map
         self.resolver_error_handler = resolver_error_handler
+
+        self.options = ConnexionOptions(old_style_options)
+        # options is added last to preserve the highest priority
+        self.options = self.options.extend(options)
+
         logger.debug('Loading specification: %s', specification,
                      extra={'swagger_yaml': specification,
-                            'base_url': base_url,
+                            'base_path': base_path,
```

Comment:
Changing `base_url` to `base_path` can break some code, it would probably be best to check first if `base_url` isn't in `**old_style_options` if `base_path` is `None` a provide a deprecation warning for a couple of versions.